tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -2505,14 +2505,38 @@ public function userListsTheResourcesSharedWithThemUsingGraphApi(string $user, s
 		}
 
 		$credentials = $this->getAdminOrUserCredentials($user);
-		$this->featureContext->setResponse(
-			GraphHelper::getSharesSharedWithMe(
+
+		// Sometimes listing shares might not return the updated shares list
+		// so try again until @client.synchronize is true for the max. number of retries (i.e. 10)
+		// and do not retry when the share is expected to be not synced
+		$tryAgain = false;
+		$retried = 0;
+		do {
+			$response = GraphHelper::getSharesSharedWithMe(
 				$this->featureContext->getBaseUrl(),
 				$this->featureContext->getStepLineRef(),
 				$credentials['username'],
 				$credentials['password']
-			)
-		);
+			);
+
+			$jsonObj = $this->featureContext->getJsonDecodedResponseBodyContent($response);
```

Comment:
```suggestion
			$jsonObj = $this->featureContext->getJsonDecodedResponseBodyContent($response);
```
$jsonObj name should be replace with meaningful name